docs/docs/plugin-authoring.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Plugin Authoring
+---
+
+Creating custom plugins in Gatsby is a breeze with our straightforward authoring process.
+
+## Core Concepts
+
+- Every Gatsby plugin is a standalone npm package
+- At minimum, a `package.json` is required
+- Plugins can be used locally (see [local plugins](#local-plugins) below) or published to npm as packages
+- A plugin has access to the the Gatsby [Node](/docs/node-apis/), [SSR](/docs/ssr-apis/), and [browser](/docs/browser-apis/) APIs
+
+## Plugin naming conventions
+
+There are four standard plugin naming conventions for Gatsby:
+
+- **`gatsby-source-*`** — a source plugin loads data from a given source (e.g. WordPress, MongoDB, the file system). Use this plugin type if you are connecting a new source of data to Gatsby.
+  - Example: `gatsby-source-contentful`
+- **`gatsby-transformer-*`** — a transformer plugin converts data from one format to another (e.g. CSV to JSON). Use this naming convention
+  - Example: `gatsby-transformer-yaml`
+- **`gatsby-[plugin-name]-*`** — if a plugin is a plugin for another plugin 😅, it should be prefixed with the name of the plugin it extends (e.g. if it adds emoji to the output of `gatsby-transformer-remark`, call it `gatsby-remark-add-emoji`). Use this naming convention whenever your plugin will be included as a plugin in the `options` object of another plugin.
+  - Example: `gatsby-remark-images`
+- **`gatsby-plugin-*`** — this is the most general plugin type. Use this naming convention if your plugin doesn’t meet the requirements of any other plugin types.
+  - Example: `gatsby-plugin-sass`
+
+## What files does Gatsby look for in a plugin?
+
+- `package.json` — [required] used to find the `name` and `version` fields (both optional)
+  - this can be an empty object (`{}`) for local plugins
+- `gatsby-browser.js` — usage details are in the [browser API reference](/docs/browser-apis/)
```

Comment:
Is it worth being explicit that these three files are optional? Similar to the _[required]_ note on `package.json`